lib/foreman/renderer/scope/macros/base.rb

```diff
@@ -137,22 +137,39 @@ def indent(count, skip1: false)
             result.join('')
           end
 
-          apipie :method, 'Performs a DNS lookup on Foreman server' do
+          apipie :method, 'Performs a Hosts and DNS lookup on Foreman server. Prefers A records over AAAA, on systems without public IPv6 addresses Ruby does not return AAAA record, use the second argument for a specific type.' do
             required :name_or_ip, String, desc: 'a hostname or IP address to perform DNS lookup for'
-            returns String, desc: 'IP resolved via DNS if hostname was given, hostname if an IP was given.'
+            optional :record_type, Symbol, desc: 'optional symbol representing record type to return: :a, :aaaa, :cname, :mx, :ptr, :txt'
+            returns String, desc: 'when no type is provided, returns IP resolved via DNS if hostname was given, hostname if an IP was given. when type is provided, performs DNS lookup of the particular type.'
             raises error: Timeout::Error, desc: 'when DNS resolve could not be performed in time set by global setting `dns_timeout`'
             raises error: Resolv::ResolvError, desc: 'when DNS resolve failed, e.g. because of misconfigured DNS server or invalid query'
             example "dns_lookup('example.com') # => '10.0.0.1'"
+            example "dns_lookup('example.com', :aaaa) # => '2001:db8:85a3:8d3:1319:8a2e:370:7348'"
             example "dns_lookup('10.0.0.1') # => 'example.com'"
-            example "echo <%= dns_lookup('example.com') %> example.com >> /etc/hosts"
+            example "echo <%= dns_lookup('example.com', :a) %> example.com >> /etc/hosts"
```

Comment:
Is this really correct? https://ruby-doc.org/stdlib-3.0.1/libdoc/resolv/rdoc/Resolv/DNS.html#method-i-getresource states:
> Returned resource is represented as a Resolv::DNS::Resource instance, i.e. Resolv::DNS::Resource::IN::A.

I guess this all relies on the `.to_s` method of the objects.

Reply:
Good catch, only those that have `DomainName` parent do have a `name` attribute, but MX and SOA are different. I am making a change: all returns `name` but TXT returns an array of values and MX only returns exchange host (and drops priority).

In all cases, we only return the FIRST record found, I think refactoring this method into something that returns all records is too much. That should be probably a different macro, this one is in use already.